query-engine/driver-adapters/src/transaction.rs

```diff
@@ -6,10 +8,78 @@ use quaint::{
     Value,
 };
 
-use crate::proxy::{TransactionOptions, TransactionProxy};
+use crate::proxy::{TransactionContextProxy, TransactionOptions, TransactionProxy};
 use crate::{proxy::CommonProxy, queryable::JsBaseQueryable, send_future::UnsafeFuture};
 use crate::{JsObject, JsResult};
 
+pub(crate) struct JsTransactionContext {
+    tx_ctx_proxy: TransactionContextProxy,
+    inner: JsBaseQueryable,
+}
+
+// Wrapper around JS transaction context objects that implements Queryable. Can be used in place of quaint transaction,
+// context, but delegates most operations to JS
+impl JsTransactionContext {
+    pub(crate) fn new(inner: JsBaseQueryable, tx_ctx_proxy: TransactionContextProxy) -> Self {
+        Self { inner, tx_ctx_proxy }
+    }
+
+    pub fn start_transaction(&self) -> impl Future<Output = quaint::Result<Box<JsTransaction>>> + '_ {
+        UnsafeFuture(self.tx_ctx_proxy.start_transaction())
+    }
+}
+
+#[async_trait]
+impl Queryable for JsTransactionContext {
```

Comment:
It's kind of a slippery slope but I really feel like implementing `Deref` is appropriate in these cases. Let's not do it in this PR to avoid churning unrelated code, but I'd strongly consider doing it later.